mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -45,37 +49,73 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(ticker.clone()))
+        .collect::<Vec<_>>();
+
+    // Get all matching orders and active swaps.
+    let mut active_swaps = vec![];
+    let mut still_matching_orders = vec![];
+    for ticker in &coins_to_disable {
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        let still_matching = try_s!(get_matching_orders(&ctx, ticker).await);
+        active_swaps.extend(swaps);
+        still_matching_orders.extend(still_matching);
     }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
+    drop_mutability!(active_swaps);
+    drop_mutability!(still_matching_orders);
+    // If there're matching orders or active swap we return this error.
+    if !active_swaps.is_empty() || !still_matching_orders.is_empty() {
         let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
+            "error": format!("There're currently matching orders or active swaps for some tokens"),
             "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
+                "active_swaps": still_matching_orders,
+                "still_matching": still_matching_orders
```

Comment:
you are using the `still_matching_orders` vec for the active swaps. Also, no need to put these 2 inside the orders key. So this should be
``` rust
        let err = json!({
            "error": format!("There're currently matching orders or active swaps for some tokens"),
            "active_swaps": active_swaps,
            "still_matching": still_matching_orders
        });
```

Reply:
done

Reply:
But doesn't this break the backward compatibility?

Reply:
Yes, It believe this does break backward compatibility for error handling in GUIs. Do you think we should keep the `cancelled` field with an empty vec and rename `still_matching` to `matching`?